executor/tests/test_executor.py

```diff
@@ -0,0 +1,417 @@
+                self.statuses.append(status)
+                if self.count_calls == 1:
+                    raise Exception('Thrown from test')
+
+            def method_call_count(self):
+                return self.count_calls
+
+        driver = FakeMesosExecutorDriver()
+        task_id = get_random_task_id()
+        task = {'task_id': {'value': task_id}}
+        stdout_name = None
+        stderr_name = None
+
+        result = ce.launch_task(driver, task, stdout_name, stderr_name)
+
+        self.assertIsNone(result)
+        self.assertEqual(2, driver.method_call_count())
+
+        actual_status_0 = driver.statuses[0]
+        expected_status_0 = {'task_id': {'value': task_id}, 'state': cook.TASK_RUNNING}
+        assert_status(expected_status_0, actual_status_0)
+
+        actual_status_1 = driver.statuses[1]
+        expected_status_1 = {'task_id': {'value': task_id}, 'state': cook.TASK_FAILED}
+        assert_status(expected_status_1, actual_status_1)
+
+    def test_cleanup_process(self):
+        task_id = get_random_task_id()
+        stdout_name = 'build/stdout.' + str(task_id)
+        stderr_name = 'build/stderr.' + str(task_id)
+
+        stdout = open(stdout_name, 'w+')
+        stderr = open(stderr_name, 'w+')
+
+        try:
+            process_info = None, stdout, stderr
+            ce.cleanup_process(process_info)
+
+            self.assertTrue(stdout.closed)
+            self.assertTrue(stderr.closed)
+        finally:
+            cleanup_output(stdout_name, stderr_name)
+
+    def test_kill_task_terminate(self):
+        driver = FakeMesosExecutorDriver()
+        task_id = get_random_task_id()
+        task = {'task_id': {'value': task_id}}
+
+        stdout_name = 'build/stdout.' + str(task_id)
+        stderr_name = 'build/stderr.' + str(task_id)
+
+        stdout = open(stdout_name, 'w+')
+        stderr = open(stderr_name, 'w+')
+
+        try:
+            command = 'sleep 100'
+            process = subprocess.Popen(command, shell=True, stdout=stdout, stderr=stderr)
+            process_info = process, stdout, stderr
+            shutdown_grace_period_ms = 2000
+            ce.kill_task(driver, task, process_info, shutdown_grace_period_ms)
+
+            # await process termination
+            while process.poll() is None:
+                time.sleep(0.01)
+            self.assertEqual(-1 * signal.SIGTERM, process.poll())
+
+            self.assertEqual(1, driver.method_call_count())
+            self.assertEqual(1, len(driver.statuses))
+            actual_status = driver.statuses[0]
+            expected_status = {'task_id': {'value': task_id},
+                               'state': cook.TASK_KILLED}
+            assert_status(expected_status, actual_status)
+
+            self.assertTrue(stdout.closed)
+            self.assertTrue(stderr.closed)
+
+        finally:
+            cleanup_output(stdout_name, stderr_name)
+
+    def test_await_process_completion(self):
```

Comment:
This test doesn't test the "await" part.  It only tests the case where `stop_signal` is set.